### PR TITLE
Swap out a few more PNGs for SVGs

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
* Swap out a few more assets for their vector representations.
* Move PaymentSheet Example snapshot tests to the StripePaymentSheetTests bundle to make it easier to re-run them.
* These tests didn't work outside of a host app (as view controller presentation seems to be broken when running directly from xctest), so reuse the StripeiOS tests host app for StripePaymentSheet's tests.
* Regenerate relevant snapshot tests.

## Testing
CI snapshots

## Changelog
None

### CODE REVIEW

File: StripePaymentsUI/StripePaymentsUI/Resources/StripePaymentsUI.xcassets/CardCarousel/card_cartes_bancaires.imageset/Contents.json [hunk 1]
```diff
@@ -1,7 +1,17 @@
 {
   "images" : [
     {
-      "filename" : "Carte Bancaire.svg",
+      "filename" : "cartesBancairesLight (1).svg",
```
Comment (yuki-stripe) [inline StripePaymentsUI/StripePaymentsUI/Resources/StripePaymentsUI.xcassets/CardCarousel/card_cartes_bancaires.imageset/Contents.json:4]: remove (1) in filename

File: StripeUICore/StripeUICore/Resources/StripeUICore.xcassets/icon_clear.imageset/Contents.json [hunk 2]
```diff
@@ -1,19 +1,8 @@
...
-    {
-      "filename" : "icon_clear@2x.png",
-      "idiom" : "universal",
-      "scale" : "2x"
-    },
-    {
-      "filename" : "icon_clear@3x.png",
-      "idiom" : "universal",
-      "scale" : "3x"
+      "filename" : "icon_clear (1).svg",
```
Comment (yuki-stripe) [inline StripeUICore/StripeUICore/Resources/StripeUICore.xcassets/icon_clear.imageset/Contents.json:4]: filename

**Review by yuki-stripe — COMMENTED** [review]
I think this is good to ship after the fixing the filename typos. Jotting down some other thoughts:
* I noticed the payment method cell width changed in the snapshot tests. I'm not sure why. Did the test device change?
* These `testPaymentSheet_LPM_*` tests feel redundant if they're composed of UI elements that are already tested.

**Comment by davidme-stripe** [comment]
> I noticed the payment method cell width changed in the snapshot tests. I'm not sure why. Did the test device change?

The tests are now running within a host app instead of XCTest, that should be the only difference. I've noticed this in other tests — we get inconsistent sizes for the width of those cells depending on the device used, PNGs vs SVGs, whether they overscroll, etc. Sometimes every cell is the same size, sometimes the cells size individually to fit their contents. I filed RUN_MOBILESDK-2677 to look into it.

File: StripePaymentsUI/StripePaymentsUI/Resources/StripePaymentsUI.xcassets/CardCarousel/card_cartes_bancaires.imageset/Contents.json (hunk 1 above)
Comment (davidme-stripe) [reply StripePaymentsUI/StripePaymentsUI/Resources/StripePaymentsUI.xcassets/CardCarousel/card_cartes_bancaires.imageset/Contents.json]: Done, also swapped these out for the latest images from JJ

File: StripeUICore/StripeUICore/Resources/StripeUICore.xcassets/icon_clear.imageset/Contents.json (hunk 2 above)
Comment (davidme-stripe) [reply StripeUICore/StripeUICore/Resources/StripeUICore.xcassets/icon_clear.imageset/Contents.json:4]: Done

**Review by yuki-stripe — APPROVED** [review]

File: StripePaymentSheet/StripePaymentSheet/Resources/StripePaymentSheet.xcassets/Carousel/carousel_applepay.imageset/Contents.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
           "value" : "dark"
         }
       ],
-      "filename" : "discover_dark.pdf",
+      "filename" : "applePayDark.svg",
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Resources/StripePaymentSheet.xcassets/Carousel/carousel_applepay.imageset/Contents.json:14]: Not really, the xcassets file is compiled to an `Assets.car` and individual assets are accessed based on the imageset name. For example:
```
  {
    "AssetType" : "Vector",
    "Colorspace" : "generic",
    "Compression" : "lzfse",
    "Idiom" : "universal",
    "Name" : "stp_card_visa",
    "NameIdentifier" : 5106,
    "RenditionName" : "icon-card-visa.svg",
    "SHA1Digest" : "38F1BFE9AE4DE2C2D86CFFFE7525F3BA0EC52C529A0D17CBCAE9EDD042F1C99E",
    "SizeOnDisk" : 1496
  }
```

**Pull request merged**